Automation rules - 2023-03-01-preview - add entity trigger API

loriatarms opened the pull request:
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. - this feature should go public around March 1st 2023.
  3. When do you expect to publish the swagger? Please provide date or, if the date is not yet available, the month. - March 1st
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

loriatarms committed:
Added entity trigger API to version 2023-03-01-preview

loriatarms committed:
Execute pretties and fixed breaking change

loriatarms committed:
Fixed example

zizw123 commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-03-01-preview/AutomationRules.json:
```diff
@@ -217,6 +217,64 @@
+          }
+        ],
+        "responses": {
+          "204": {
```
>204 [](http://example.com/codeflow?start=11&length=3)

Why is this 204? If action is completed successfully, it should return 200

loriatarms replied:
This action is calling LogicApps API to trigger a playbook on an entity. It returns 204 since we know that the invocation was "accepted" but we don't know whether it was completed successfully.
The response of this endpoint aligns with the response code in another similar endpoint that we have - "/incidents/{incidentID}/runPlaybook" that returns 204 for the same reason and it has existed for a long time now.

zizw123 commented:
Please also check https://github.com/Azure/azure-rest-api-specs/pull/22523/checks?check_run_id=11226466477

loriatarms commented:
Regarding the LintDiff validation:
1.	The endpoint I have added has an optional parameter of tenant ID which is always a GUID.
2.	The format of the tenant ID aligns with existing endpoints across our API – for example “incident/{incidentID}/runPlaybook” – a similar endpoint to the one I’ve created that trigger a playbook on an incident instead of an entity:
[https://github.com/Azure/azure-rest-api-specs/blob/59780936a881bd149930f5d8f479bc24896a47ab/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-03-01-preview/AutomationRules.json#:~:text=%22properties%22%3A%20%7B-,%22tenantId%22%3A%20%7B,%7D%2C,-%22logicAppsResourceId%22%3A%20%7B](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fblob%2F59780936a881bd149930f5d8f479bc24896a47ab%2Fspecification%2Fsecurityinsights%2Fresource-manager%2FMicrosoft.SecurityInsights%2Fpreview%2F2023-03-01-preview%2FAutomationRules.json%23%3A~%3Atext%3D%2522properties%2522%253A%2520%257B-%2C%2522tenantId%2522%253A%2520%257B%2C%257D%252C%2C-%2522logicAppsResourceId%2522%253A%2520%257B&data=05%7C01%7Cloriatar%40microsoft.com%7Cdb150290c3b24ed542bb08db0f8b5fe8%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638120863527832427%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=uolKZDcaijGk2qx5RpKuYNVVjp3r6ADIKT8tv5BdRaM%3D&reserved=0)

There are two reasons why we need this tenant ID as a separate parameter and can’t only work with the one passed in the header from ARM:
1)	Consistency – we have another similar endpoint that triggers a playbook on an incident – Incidents/{incidentId}/runPlaybook (the one I’ve added can trigger playbooks on entities).
To supply the best experience to our end-users, we should always try to be aligned in such scenarios so if one can get the tenant ID as a parameter, the other one should as well.
2)	Technical reasons – the tenant ID of the user will not always be the same as the tenant of his workspace / LogicApp workflow.
It relevant both for portal users, where for example lighthouse users can see multiple tenants at the same view so the tenant of the incident/entity will not always be the same as the tenant of the LogicApp workflow and for users who are using this feature directly from our API.
Eventually, we use the given tenant ID to create an AAD token for the tenant of the LogicApp workflow so as I said, it will not always be identical to the user’s current tenant / to the tenant of the incident/entity.

NetRock commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-03-01-preview/AutomationRules.json:
```diff
@@ -217,6 +217,64 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/entities/{entityIdentifier}/runPlaybook": {
```
need to add new api-version for new features, since this is public facing api.

loriatarms committed:
Added descriptions
